windows-agent/internal/proservices/landscape/executor_test.go

```diff
@@ -148,47 +148,40 @@ func TestReceiveCommandStartStop(t *testing.T) {
 func TestInstall(t *testing.T) {
 	t.Parallel()
 
-	ctx, cancel := context.WithCancel(context.Background())
-	t.Cleanup(cancel)
-
-	fileServerAddr := mockRootfsFileServer(t, ctx)
-
-	emptyFileChecksum := "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
-	mockErrorChecksum := "afe55cda4210c2439b47c62c01039027522f7ed4abdb113972b3030b3359532a"
-	mockMismatchChecksum := "1234"
-
 	testCases := map[string]struct {
 		noCloudInit            bool
 		cloudInitWriteErr      bool
 		distroAlreadyInstalled bool
-		emptyDistroName        bool
+		distroName             interface{} // using interface{} to allow nil, modelling an optional value.
 		wslInstallErr          bool
 		appxDoesNotExist       bool
 		nonResponsiveServer    bool
 		breakVhdxDir           bool
 		breakTarFile           bool
 		breakTempDir           bool
 
-		sendRootfsURL      string
-		sendRootfsChecksum string
+		sendRootfsURL    string
+		missingChecksums bool
 
 		wantCouldInitWriteCalled bool
 		wantInstalled            bool
 	}{
 		"From the store":                    {wantInstalled: true, wantCouldInitWriteCalled: true},
-		"From a rootfs URL":                 {sendRootfsURL: "goodfile", wantInstalled: true},
-		"From a rootfs URL with a checksum": {sendRootfsURL: "goodfile", sendRootfsChecksum: emptyFileChecksum, wantInstalled: true},
+		"From a rootfs URL with a checksum": {sendRootfsURL: "goodfile", wantInstalled: true},
 		"With no cloud-init":                {noCloudInit: true, wantCouldInitWriteCalled: true, wantInstalled: true},
 
-		"Error when the distroname is empty":         {emptyDistroName: true},
 		"Error when the Appx does not exist":         {appxDoesNotExist: true},
 		"Error when the distro is already installed": {distroAlreadyInstalled: true, wantInstalled: true},
 		"Error when the distro fails to install":     {wslInstallErr: true},
 		"Error when cannot write cloud-init file":    {cloudInitWriteErr: true, wantCouldInitWriteCalled: true},
+		"Error when the distroname is empty":         {distroName: ""},
 
-		"Error when the rootfs isn't a valid tarball":                   {sendRootfsURL: "badfile", sendRootfsChecksum: mockErrorChecksum, wantInstalled: false},
-		"Error when the checksum doesn't match":                         {sendRootfsURL: "goodfile", sendRootfsChecksum: mockMismatchChecksum, wantInstalled: false},
+		"Error when the distro ID is reserved":                          {sendRootfsURL: "goodfile", distroName: "Ubuntu-19.13", wantInstalled: false},
+		"Error when the rootfs isn't a valid tarball":                   {sendRootfsURL: "badfile", wantInstalled: false},
+		"Error when the checksum file doesn't exist":                    {missingChecksums: true, sendRootfsURL: "goodfile", wantInstalled: false},
```

Comment:
I think we shouldn’t require the checksum fail, just issue a warning if we couldn’t find one (maybe we should differentiate 404 with other errors).

However, there is a test case that if there is a checksum file, it should list that rootfs or error out (you can have this with no extra line by return "" with no default if not found. Then, we would compare against this and "oh, no match because it’s empty" (which really means there is no entry)

Reply:
Well, since the checksums file is optional, we should `return "", nil` if that file is not found but `return "", ERROR` if the file exists but does not contain an entry for the rootfs being checked in `wantRootfsChecksum`, this way we know and react accordingly if the checksum string is empty.

Reply:
Sorry, I wrote warning here, but it should be info looking at the context, just wrote a comment on it :)

Agreed on the comment!